cmd/ltagent/init.go

```diff
@@ -110,23 +119,43 @@ func RunInitCmdF(cmd *cobra.Command, args []string) error {
 
 	config.UserControllerConfiguration.Type = loadtest.UserControllerGenerative
 	config.UsersConfiguration.InitialActiveUsers = 0
+	config.UsersConfiguration.PercentageOfAdminUsers = adminPercent
 	config.UserControllerConfiguration.RatesDistribution = []loadtest.RatesDistribution{
 		{
 			Rate:       0.2,
 			Percentage: 1.0,
 		},
 	}
 
-	newC, err := api.NewControllerWrapper(config, &genConfig, 0, userPrefix, nil)
-	if err != nil {
-		return fmt.Errorf("error while creating new controller: %w", err)
+	adminUserCount := int64(config.UsersConfiguration.PercentageOfAdminUsers * float64(initialUserCount))
+	data := map[string]int64{
+		userentity.AdminPrefix: adminUserCount,
+		userPrefix:             initialUserCount - adminUserCount,
 	}
-	lt, err := loadtest.New(config, newC, log)
-	if err != nil {
-		return fmt.Errorf("error while initializing loadtest: %w", err)
+
+	return genDataWithPrefix(config, genConfig, log, data)
+}
+
+func genDataWithPrefix(config *loadtest.Config, genConfig gencontroller.Config, logger *mlog.Logger, seedData map[string]int64) error {
+	for userPrefix, numUsers := range seedData {
+		newC, err := api.NewControllerWrapper(config, &genConfig, 0, userPrefix, nil)
+		if err != nil {
+			return fmt.Errorf("error while creating new controller: %w", err)
+		}
+
+		lt, err := loadtest.New(config, newC, logger)
+		if err != nil {
+			return fmt.Errorf("error while initializing loadtest: %w", err)
+		}
+
+		mlog.Info(fmt.Sprintf("generating %d users with prefix %s", numUsers, userPrefix))
+		err = genData(lt, numUsers)
+		if err != nil {
+			return err
+		}
```

Comment:
I am not sure this is going to work as we expect it. We shouldn't be running the generative load-test more than once.
I think for this case we might want to keep it very simple and before even running this make sure the configured number of admins gets created without the need to going through the load-test. Just creating some user entities and promoting them in a loop will do.

Reply:
I've tried to keep things as homogenous as possible. Happy to have the admins being generated up front. Personally, I do not see a problem with running generative loadTest instances since the init phase is a one-off operation, no strong pref though.

Reply:
The main issue I see doing this is that the admin users will end up generating all the data. The way the gencontroller works is that it's not going to stop until all data has been generated.

Reply:
Yeah true, which in a sense is ok right, since essentially what are we doing above is splitting the genData onto two segments, one with N users and one with M admins but our intention still remains to generate N+M users.

Reply:
The point is generating the data (as in teams, channels, posts, reactions), not the users. So the second pass would not really be needed if the admins already created everything. But also there was a reason those numbers were chosen (50 at 0.2 rate). Those values gave the highest throughput.

Reply:
Right, I didn't know about the second reason. Cool, that makes sense. Will amend.